src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +88,16 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        Predicate<Person> predicate = p -> 
+                (p.getSex() == Person.Sex.WOMAN)
+                && (p.getAge() >= femaleAge)
+                && (p.getCats() != null);
+        return peopleList.stream()
+                .filter(predicate)
+                .map(p -> p.getCats())
+                .flatMap(x -> x.stream())
+                .map(x -> x.getName())
```

Comment:
You can simplify this
```suggestion
                .flatMap(person -> person.getCats().stream())
                .map(Cat::getName)
```